src/viem/prepareTransactions.ts

```diff
@@ -61,6 +69,24 @@ export function getMaxGasFee(txs: TransactionRequest[]): BigNumber {
   return new BigNumber(maxGasFee.toString())
 }
 
+export function getEstimatedGasFee(txs: TransactionRequest[]): BigNumber {
+  let estimatedGasFee = BigInt(0)
+  for (const tx of txs) {
+    // Use _estimatedGasUse if available, otherwise use gas
+    const estimatedGas = tx._estimatedGasUse ?? tx.gas
+    if (!estimatedGas) {
+      throw new Error('Missing _estimatedGasUse or gas')
+    }
+    if (!tx._baseFeePerGas || !tx.maxFeePerGas) {
+      throw new Error('Missing _baseFeePerGas or maxFeePerGas')
+    }
+    const expectedFeePerGas = tx._baseFeePerGas + (tx.maxPriorityFeePerGas ?? BigInt(0))
+    estimatedGasFee +=
+      estimatedGas * (expectedFeePerGas < tx.maxFeePerGas ? expectedFeePerGas : tx.maxFeePerGas)
+  }
+  return new BigNumber(estimatedGasFee.toString())
+}
```

Comment:
The formula implemented is this:

`estimatedGasFee = (estimatedGasUse ?? gas) * min(baseFeePerGas + (maxPriorityFeePerGas ?? 0), maxFeePerGas)`